### PR TITLE
rpcserver: add handleSendRawTransaction tests.

### PR DESCRIPTION
**Pull request by dnldd**
**depends on #2405**

This adds rpc tests for `handleSendRawTransaction`.

Work towards #2069.

### CODE REVIEW

**Comment by dnldd** [comment]
```sh
go tool cover -func=cov.out | grep -E "handleSendRawTransaction"
github.com/decred/dcrd/internal/rpcserver/rpcserver.go:4505:            handleSendRawTransaction      97.1%
```

The deserialization error [here](https://github.com/decred/dcrd/blob/69e46f99cc386fd0e1d948721597ca9fcbd5d7d7/internal/rpcserver/rpcserver.go#L4521) could not be hit.

File: internal/rpcserver/interface.go [hunk 1]
```diff
@@ -654,3 +654,128 @@ type TxIndexer interface {
...
+
+	// NotifyWinningTickets passes newly winning tickets to the manager for
+	// processing.
+	NotifyWinningTickets(wtnd *WinningTicketsNtfnData)
+
+	// NotifySpentAndMissedTickets passes ticket spend and missing data for an
+	// incoming block to the manager for processing.
+	NotifySpentAndMissedTickets(tnd *blockchain.TicketNotificationsData)
+
+	// NotifyNewTickets passes a new ticket data for an incoming block to the
```
Comment (rstaudt2) [inline internal/rpcserver/interface.go:691]:
```suggestion
	// NotifyNewTickets passes new ticket data for an incoming block to the
```

**Review by rstaudt2 — COMMENTED** [review]
Looks good overall, just had a few comments.

**Review by davecgh — APPROVED** [review]
I guess I'll not hold up this PR since it is primarily test code and things are already structured this way, but the fact this is introducing such a huge interface is a very clear sign that things are not separately out properly.  This is _especially_ true because the interface has a `Run` method on it.  That is a major red flag.

**Comment by rstaudt2** [comment]
> I guess I'll not hold up this PR since it is primarily test code and things are already structured this way, but the fact this is introducing such a huge interface is a very clear sign that things are not separately out properly. This is _especially_ true because the interface has a `Run` method on it. That is a major red flag.

I was thinking the same but didn't see an easy way to separate it out, so this seemed like a good first step to get better coverage in.  Ideally, I think we would want these tests to validate that notifications are queued as expected (rather than just stubbing those methods to do nothing) and have separate tests to confirm that notifications are processed off the queue as expected by the notification manager.

**Review by rstaudt2 — APPROVED** [review]

**Comment by dnldd** [comment]
> I guess I'll not hold up this PR since it is primarily test code and things are already structured this way, but the fact this is introducing such a huge interface is a very clear sign that things are not separately out properly. This is _especially_ true because the interface has a `Run` method on it. That is a major red flag.

Noted, I'll start looking into simplifying that interface in a separate PR.

**Comment by davecgh** [comment]
>  I think we would want these tests to validate that notifications are queued as expected (rather than just stubbing those methods to do nothing) and have separate tests to confirm that notifications are processed off the queue as expected by the notification manager.

Agreed.  I would expect something more like a mock websocket client that actually listens for the expected notifications with appropriate timeouts to ensure they're being queued and sent properly.

That aside, this needs a rebase to resolve merge conflicts and then I'll get it merged.

**Pull request merged**